core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -167,39 +179,104 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
+                attackableTile = battleHelper
                         .getAttackableEnemies(attacker.unit, attacker.unit.movement.getDistanceToTiles())
                         .firstOrNull{ it.tileToAttack == defender.getTile()}
             }
             else if (attacker is CityCombatant)
             {
                 val canBombard = UnitAutomation().getBombardTargets(attacker.city).contains(defender.getTile())
                 if (canBombard) {
-                    attackableEnemy = AttackableTile(attacker.getTile(), defender.getTile())
+                    attackableTile = AttackableTile(attacker.getTile(), defender.getTile())
                 }
             }
         }
 
-        if (!worldScreen.isPlayersTurn || attackableEnemy == null) {
+        if (!worldScreen.isPlayersTurn || attackableTile == null) {
             attackButton.disable()
             attackButton.label.color = Color.GRAY
         }
 
         else {
             attackButton.onClick {
                 try {
-                    Battle.moveAndAttack(attacker, attackableEnemy)
+                    Battle.moveAndAttack(attacker, attackableTile)
+                    worldScreen.mapHolder.unitActionOverlay?.remove() // the overlay was one of attacking
+                    worldScreen.shouldUpdate = true
+                }
+                catch (ex:Exception){
+                    val battleBugPopup = Popup(worldScreen)
+                    battleBugPopup.addGoodSizedLabel("You've encountered a bug that I've been looking for for a while!").row()
+                    battleBugPopup.addGoodSizedLabel("If you could copy your game data (\"Copy saved game to clipboard\" - ").row()
+                    battleBugPopup.addGoodSizedLabel("  paste into an email to yairm210@hotmail.com)").row()
+                    battleBugPopup.addGoodSizedLabel("It would help me figure out what went wrong, since this isn't supposed to happen!").row()
+                    battleBugPopup.addGoodSizedLabel("If you could tell me which unit was selected and which unit you tried to attack,").row()
+                    battleBugPopup.addGoodSizedLabel("  that would be even better!").row()
+                    battleBugPopup.open()
+                }
+            }
+        }
+
+        add(attackButton).colspan(2)
+
+        pack()
+
+        setPosition(worldScreen.stage.width/2-width/2, 5f)
+    }
+
+    private fun simulateNuke(attacker: MapUnitCombatant, targetTile: TileInfo){
+        clear()
+        defaults().pad(5f)
+
+        val attackerNameWrapper = Table()
+        val attackerLabel = attacker.getName().toLabel()
+        attackerNameWrapper.add(UnitGroup(attacker.unit,25f)).padRight(5f)
+        attackerNameWrapper.add(attackerLabel)
+        add(attackerNameWrapper)
+
+        val defenderNameWrapper = Table()
+        for (tile in targetTile.getTilesInDistance(Battle.NUKE_RADIUS)) {
+            val defender: ICombatant = tryGetDefenderAtTile(tile, true) ?: continue
```

Comment:
don't :? continue, split that into a separate line, if(defender==null) continue

Reply:
Yes, that's right, `if(defender==null)` then `continue`.
I assume `:? continue` is more compact way of writing the code: 1 line instead of 2.
If you believe having 2 lines is better, I will correct that.

Reply:
I have removed the "Elvis" operator, however, I still believe it is useful here, as Google recommends it as a common Kotlin pattern - https://developer.android.com/kotlin/common-patterns#handle-nullability